sweep/README.md

```diff
@@ -0,0 +1,185 @@
+returning fee rate at the ending fee rate.
+
+The starting fee rate is the estimated fee rate using the deadline as the conf
+target, which can be overridden by setting the `--sat_per_vbyte` via `bumpfee`
+cli. The ending fee rate is the value from `budget/tx_vbyte`, which can be
+overridden by setting the `--budget` via `bumpfee` cli.
+
+## Sweeping Outputs from a Force Close Transaction
+
+A force close transaction may have the following outputs,
+
+- Commit outputs, which are the  `to_local` and `to_remote` outputs.
+- HTLC outputs, which are the  `incoming_htlc` and `outgoing_htlc` outputs.
+- Anchor outputs, which are the local and remote anchor outputs.
+
+#### Sweeping Commit Outputs
+
+The only output we can spend is the `to_local` output. Because it can only be
+spent using our signature, there’s no time pressure here. By default, the
+sweeper will use a deadline of 1008 blocks as the confirmation target for
+non-time-sensitive outputs.  To overwrite the default, users can specify a
+value using the config `--sweeper.nodeadlineconftarget`.
+
+To specify the budget, users can use `--sweeper.budget.tolocal` to set the max
+allowed fees in sats, or use `--sweeper.budget.tolocalratio` to set a
+proportion of the `to_local` value to be used as the budget.
+
+#### Sweeping HTLC Outputs
+
+When facing a local force close transaction, HTLCs are spent in a two-stage
+setup - the first stage is to spend the outputs using pre-signed HTLC
+success/timeout transactions, the second stage is to spend the outputs from
+these success/timeout transactions. All these outputs are automatically handled
+by `lnd`, in specific,
+- For an incoming HTLC in stage one, the deadline is specified using its CLTV
+  from the timeout path. This output is time-sensitive.
+- For an outgoing HTLC in stage one, the deadline is derived from its
+  corresponding incoming HTLC’s CLTV. This output is time-sensitive.
+- For both incoming and outgoing HTLCs in stage two, because they can only be
+  spent by us, there is no time pressure to confirm them under a deadline.
+
+When facing a remote force close transaction, HTLCs can be directly spent from
+this transaction, and they are both time-sensitive outputs.
+
+By default, `lnd` will use 50% of the HTLC value as its budget. To customize
+it, users can specify `--sweeper.budget.deadlinehtlc` and
+`--sweeper.budget.deadlinehtlcratio` for time-sensitive HTLCs, and
+`--sweeper.budget.nodeadlinehtlc` and `--sweeper.budget.nodeadlinehtlcratio`
+for non-time-sensitive sweeps.
+
+#### Sweeping Anchor Outputs
+
+An anchor output is a special output that functions as “anchor” to speed up the
+unconfirmed force closing transaction via CPFP. Most of the time, anchor
+outputs won’t be swept due to it’s uneconomical. However, if the force close
+transaction contains time-sensitive outputs (HTLCs), and stays unconfirmed, the
+anchor output will be swept to accelerate the force close process.
```

Comment:
> Most of the time, anchor outputs won't be swept...

Is this true?  Probably we don't need to mention frequency at all.  For example:

```suggestion
An anchor output is a special output that functions as “anchor” to speed up the
unconfirmed force closing transaction via CPFP. If the force close transaction doesn't contain any HTLCs, the anchor
output is generally uneconomical to sweep and will be ignored. However, if the force close
transaction does contain time-sensitive outputs (HTLCs), the
anchor output will be swept to CPFP the transaction and accelerate the force close process.
```

Reply:
thanks and updated!

>Is this true?

Yeah with the new sweeper, unless it's for CPFP, the anchor output won't be aggregated at all due to it comes with a budget (330) that cannot cover it's min relay fee.